tools/c2chapel/c2chapel.py

```diff
@@ -370,7 +370,11 @@ def genStructOrUnion(structOrUnion, name=""):
         print()
         return
 
-    ret = "extern union " if isUnion else "extern record ";
+    ret = ""
+    if isAnon:
+        ret = "extern union " if isUnion else "extern record "
+    else:
+        ret = "extern union " if isUnion else "extern \"struct " + name + "\" record "
```

Comment:
Seems like this should do a similar thing (`extern "union myunion" union`) and I more or less expect that doesn't work today but it'd be good to create an issue about it

Reply:
After testing it a bit, it seems that it does work for unions and in the same way that it does for structs (i.e., we need to generate the `extern "union myunion" union`). Putting this change in and updating the testing then will re-push. Does that sound good to you or is there something I may not be anticipating here that you would prefer an issue be open for?

I would note that the behavior is different for LLVM and C backend, but it doesn't seem to be a problem. The LLVM backend lets you always refer to union with the `extern "union myunion" union` syntax even when it is typedef'd, where the C backend will give an error on the exact same code.

Reply:
As long as you check with both C and LLVM code generation that it works.